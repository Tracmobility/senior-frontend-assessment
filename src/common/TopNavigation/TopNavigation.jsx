import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faSearch,  faBell, faEnvelope } from '@fortawesome/free-solid-svg-icons';
import profileImg from '../../assets/img/profileImg.jpeg';
import styles from './TopNavigation.module.scss';

const TopNavigationComponent = () => (
  <header className={styles.topNavigationContainer}>
    <nav>
      <div className={styles.topNavSearch}>
        <input type="text" placeholder="Search For.." />
        <button>
          <FontAwesomeIcon icon={faSearch}/>
        </button>
      </div>
      <div className={styles.topNavProfileButtons}>
        <button>
          <FontAwesomeIcon icon={faBell} />
          <p>4</p>
        </button>
        <button>
          <FontAwesomeIcon icon={faEnvelope} />
          <p>1</p>
        </button>
      </div>
      <div className={styles.topNavUserProfile}>
        <p>Jack Niu</p>
        <img src={profileImg} alt="Jack Niu's Profile" />
      </div>
    </nav>
  </header>
);

export default TopNavigationComponent;
