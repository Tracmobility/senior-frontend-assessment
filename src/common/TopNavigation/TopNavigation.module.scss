.topNavigationContainer {
  position: absolute;
  top: 0;
  right: 0;
  width: 100%;
  height: 8%;
  background-color: #fefefe;
  border-bottom: 1px solid #d4d4d4;
  -webkit-box-shadow: 0 0 10px 1px rgba(127, 140, 141, 0.6);
  -moz-box-shadow: 0 0 10px 1px rgba(127, 140, 141, 0.6);
  box-shadow: 0 0 10px 1px rgba(127, 140, 141, 0.6);
  z-index: 1;

  nav {
    display: flex;
    width: 100%;
    flex-flow: row nowrap;
    align-items: center;
    padding: 20px;

    .topNavSearch {
      width: 28%;

      input,
      button {
        height: 35px;
      }

      input {
        width: 90%;
        background-color: #f2f2f2;
        border: none;
        outline: none;
        padding: 0 10px;
        border-top-left-radius: 4px;
        border-bottom-left-radius: 4px;

        &::placeholder {
          font-size: 14px;
          color: #7f8c8d;
        }
      }

      button {
        width: 10%;
        outline: none;
        border: none;
        background-color: #03989e;
        color: #fefefe;
        border-top-right-radius: 8px;
        border-bottom-right-radius: 8px;
      }
    }

    .topNavUserProfile {
      display: flex;
      flex-flow: row nowrap;
      align-items: center;
      padding-left: 20px;

      p {
        margin-right: 20px;
        margin-bottom: 0;
        color: #7f8c8d;
      }

      img {
        height: 40px;
        border-radius: 50%;
      }
    }

    .topNavProfileButtons {
      display: flex;
      flex-flow: row nowrap;
      align-items: center;
      width: auto;
      margin-left: auto;
      border-right: 1px solid #ccd1d1;

      button {
        width: 40px;
        display: flex;
        flex-flow: row nowrap;
        position: relative;
        border: none;
        outline: none;
        background-color: transparent;
        color: #959595;
        padding: 10px;
        margin-right: 20px;

        svg {
          font-size: 18px;
        }

        p {
          font-size: 12px;
          position: absolute;
          top: 2px;
          right: 4px;
          background-color: rgba(231, 76, 60, 0.9);
          padding: 1px 4px;
          margin-bottom: 0;
          color: #fefefe;
          border-radius: 3px;
        }
      }
    }
  }
}
