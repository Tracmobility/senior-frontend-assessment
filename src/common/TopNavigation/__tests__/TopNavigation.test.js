import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import Enzyme, { mount } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import TopNavigation from '../TopNavigation';

Enzyme.configure({ adapter: new Adapter() });

const topNavigation = (<TopNavigation />);

describe('The appearance and functionality of the top navigation component', () => {
  const container = mount(topNavigation);

  it('Matches the previous snapshot of the Top Navigation Component', () => {
    const tree = renderer.create(topNavigation).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Contains the correct master container with the correct attributes', () => {
    expect(container.childAt(0).name()).toBe('header');
    expect(container.childAt(0).hasClass('topNavigationContainer')).toBe(true);
  });

  it('Contains the search bar', () => {
    const navSearch = container.find('.topNavSearch');
    expect(navSearch.childAt(0).name()).toBe('input');
    expect(navSearch.childAt(0).props().placeholder).toBe('Search For..');
    expect(navSearch.childAt(1).name()).toBe('button');
  });

  it('Contains the notification and message buttons', () => {
    const navProfileButtons = container.find('.topNavProfileButtons');
    expect(navProfileButtons.childAt(0).name()).toBe('button');
    expect(navProfileButtons.childAt(0).childAt(0).name()).toBe('FontAwesomeIcon');
    expect(navProfileButtons.childAt(1).name()).toBe('button');
    expect(navProfileButtons.childAt(1).childAt(0).name()).toBe('FontAwesomeIcon');
  });

  it('Should display the users profile image and name', () => {
    const userProfileSection = container.find('.topNavUserProfile');
    expect(userProfileSection.childAt(0).text()).not.toBe(undefined);
    expect(userProfileSection.childAt(1).prop('src')).toBe('profileImg.jpeg');
  });
})