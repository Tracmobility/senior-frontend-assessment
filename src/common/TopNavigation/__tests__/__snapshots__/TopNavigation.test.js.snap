// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The appearance and functionality of the top navigation component Matches the previous snapshot of the Top Navigation Component 1`] = `
<header
  className="topNavigationContainer"
>
  <nav>
    <div
      className="topNavSearch"
    >
      <input
        placeholder="Search For.."
        type="text"
      />
      <button>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-search fa-w-16 "
          data-icon="search"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </button>
    </div>
    <div
      className="topNavProfileButtons"
    >
      <button>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-bell fa-w-14 "
          data-icon="bell"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M224 512c35.32 0 63.97-28.65 63.97-64H160.03c0 35.35 28.65 64 63.97 64zm215.39-149.71c-19.32-20.76-55.47-51.99-55.47-154.29 0-77.7-54.48-139.9-127.94-155.16V32c0-17.67-14.32-32-31.98-32s-31.98 14.33-31.98 32v20.84C118.56 68.1 64.08 130.3 64.08 208c0 102.3-36.15 133.53-55.47 154.29-6 6.45-8.66 14.16-8.61 21.71.11 16.4 12.98 32 32.1 32h383.8c19.12 0 32-15.6 32.1-32 .05-7.55-2.61-15.27-8.61-21.71z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        <p>
          4
        </p>
      </button>
      <button>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-envelope fa-w-16 "
          data-icon="envelope"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M502.3 190.8c3.9-3.1 9.7-.2 9.7 4.7V400c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V195.6c0-5 5.7-7.8 9.7-4.7 22.4 17.4 52.1 39.5 154.1 113.6 21.1 15.4 56.7 47.8 92.2 47.6 35.7.3 72-32.8 92.3-47.6 102-74.1 131.6-96.3 154-113.7zM256 320c23.2.4 56.6-29.2 73.4-41.4 132.7-96.3 142.8-104.7 173.4-128.7 5.8-4.5 9.2-11.5 9.2-18.9v-19c0-26.5-21.5-48-48-48H48C21.5 64 0 85.5 0 112v19c0 7.4 3.4 14.3 9.2 18.9 30.6 23.9 40.7 32.4 173.4 128.7 16.8 12.2 50.2 41.8 73.4 41.4z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        <p>
          1
        </p>
      </button>
    </div>
    <div
      className="topNavUserProfile"
    >
      <p>
        Jack Niu
      </p>
      <img
        alt="Jack Niu's Profile"
        src="profileImg.jpeg"
      />
    </div>
  </nav>
</header>
`;
